cli/pkg/cmdutil/githelper.go

```diff
@@ -0,0 +1,207 @@
+}
+
+func newGitHelper(h *Helper, org, project, localPath string) *GitHelper {
+	return &GitHelper{
+		h:           h,
+		org:         org,
+		project:     project,
+		localPath:   localPath,
+		gitConfigMu: semaphore.NewWeighted(1),
+	}
+}
+
+func (g *GitHelper) GitConfig(ctx context.Context) (*gitutil.Config, error) {
+	err := g.gitConfigMu.Acquire(ctx, 1)
+	if err != nil {
+		return nil, err
+	}
+	defer g.gitConfigMu.Release(1)
+	if g.gitConfig != nil && !g.gitConfig.IsExpired() {
+		return g.gitConfig, nil
+	}
+
+	c, err := g.h.Client()
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.GetCloneCredentials(ctx, &adminv1.GetCloneCredentialsRequest{
+		Organization: g.org,
+		Project:      g.project,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if resp.GitRepoUrl == "" {
+		return nil, fmt.Errorf("project %q is not connected to a git repository", g.project)
+	}
+	g.gitConfig = &gitutil.Config{
+		Remote:            resp.GitRepoUrl,
+		Username:          resp.GitUsername,
+		Password:          resp.GitPassword,
+		PasswordExpiresAt: resp.GitPasswordExpiresAt.AsTime(),
+		DefaultBranch:     resp.GitProdBranch,
+		Subpath:           resp.GitSubpath,
+	}
+	return g.gitConfig, nil
+}
+
+func (g *GitHelper) PushToNewManagedRepo(ctx context.Context) (*adminv1.CreateManagedGitRepoResponse, error) {
+	c, err := g.h.Client()
+	if err != nil {
+		return nil, err
+	}
+
+	gitRepo, err := c.CreateManagedGitRepo(ctx, &adminv1.CreateManagedGitRepoRequest{
+		Organization: g.org,
+		Name:         g.project,
+	})
+	if err != nil {
+		return nil, err
+	}
+	author, err := AutoCommitGitSignature(ctx, g.h, g.localPath)
+	if err != nil {
+		return nil, err
+	}
+	err = gitutil.CommitAndForcePush(ctx, g.localPath, gitRepo.Remote, gitRepo.Username, gitRepo.Password, gitRepo.DefaultBranch, author)
+	if err != nil {
+		return nil, err
+	}
+
+	err = g.setGitConfig(ctx, &gitutil.Config{
+		Remote:            gitRepo.Remote,
+		Username:          gitRepo.Username,
+		Password:          gitRepo.Password,
+		PasswordExpiresAt: gitRepo.PasswordExpiresAt.AsTime(),
+		DefaultBranch:     gitRepo.DefaultBranch,
+		Subpath:           "",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return gitRepo, nil
+}
+
+func (g *GitHelper) PushToManagedRepo(ctx context.Context) error {
+	gitConfig, err := g.GitConfig(ctx)
+	if err != nil {
+		return err
+	}
+
+	author, err := AutoCommitGitSignature(ctx, g.h, g.localPath)
+	if err != nil {
+		return err
+	}
+	err = gitutil.CommitAndForcePush(ctx, g.localPath, gitConfig.Remote, gitConfig.Username, gitConfig.Password, gitConfig.DefaultBranch, author)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (g *GitHelper) setGitConfig(ctx context.Context, c *gitutil.Config) error {
+	err := g.gitConfigMu.Acquire(ctx, 1)
+	if err != nil {
+		return err
+	}
+	defer g.gitConfigMu.Release(1)
+
+	g.gitConfig = c
+	return nil
+}
+
+func AutoCommitGitSignature(ctx context.Context, h *Helper, path string) (*object.Signature, error) {
```

Comment:
Maybe `h.GitSignature(ctx, path)`? The `*Helper` is usually accessed, not passed.